kafkajobs/listener.go

```diff
@@ -137,24 +138,49 @@ func (d *Driver) listen() error {
 			}
 		}
 
-		fetches.EachRecord(func(r *kgo.Record) {
-			item := fromConsumer(r, d.requeueCh, d.recordsCh, &d.stopped)
-
-			ctxT, span := d.tracer.Tracer(tracerName).Start(otel.GetTextMapPropagator().Extract(context.Background(), propagation.HeaderCarrier(item.headers)), "kafka_listener")
-			d.prop.Inject(ctxT, propagation.HeaderCarrier(item.headers))
-
-			d.pq.Insert(item)
-
-			span.End()
-		})
+		switch d.cfg.ConsumerOpts.PipeliningStrategy {
+		case SerialPipelining:
+			fetchWg := &sync.WaitGroup{}
+			fetches.EachPartition(func(partition kgo.FetchTopicPartition) {
+				fetchWg.Add(1)
+				itemWg := &sync.WaitGroup{}
+
+				go func() {
```

Comment:
In Go 1.25 (which is used in the project), you may use itemWg.Go to start a goroutine and add +1 to the itemWg automatically.